scripts/zkai_attention_derived_d128_rmsnorm_public_row_gate.py

```
@@ -0,0 +1,525 @@
+    if payload.get("payload_commitment") != payload_commitment(payload):
+        raise AttentionDerivedD128RmsnormPublicRowError("payload commitment drift")
+
+
+def validate_payload(payload: Any, *, context: dict[str, Any] | None = None) -> None:
+    data = _dict(payload, "payload")
+    context = build_context() if context is None else context
+    validate_core_payload(data, context=context)
+    if set(data) == FINAL_KEYS:
+        cases = _list(data.get("cases"), "cases")
+        if data.get("mutation_inventory") != list(EXPECTED_MUTATIONS):
+            raise AttentionDerivedD128RmsnormPublicRowError("mutation inventory drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)